hasher-matcher-actioner/hmalib/lambdas/fetcher.py

```diff
@@ -182,38 +184,55 @@ def __init__(
         s3_bucket: t.Any,  # Not typable?
         s3_te_data_folder: str,
         data_store_table: str,
+        supported_signal_types: t.List[SignalType],
     ) -> None:
         super().__init__(privacy_group)
         self.app_id = app_id
         self._cached_state: t.Optional[t.Dict] = None
         self.s3_bucket = s3_bucket
         self.s3_te_data_folder = s3_te_data_folder
         self.data_store_table = data_store_table
+        self.supported_signal_types = supported_signal_types
+
+    @functools.lru_cache(maxsize=None)
+    def _get_supported_indicator_types(self):
```

Comment:
nit: Maybe add this to the constructor before setting supported_signal_types. That way we know its only called once and no one uses the raws `self.supported_signal_types` instead of this function

Reply:
That is a great idea!

Reply:
Pushed in https://github.com/facebook/ThreatExchange/pull/728/commits/5bfd877dbe8478085a014e61b94c0631bd07733a